docs/CONNECTOR_PROTOCOL.md

```diff
@@ -22,120 +21,72 @@ At this stage, our assumption is that one connector will manage one index, and o
-    reduce_whitespace: boolean; -> Whether or not the `request_pipeline` should squish redundant whitespace
-    run_ml_inference: boolean; -> Whether or not the `request_pipeline` should run the ML Inference pipeline
+    reduce_whitespace: boolean; -> Whether the `request_pipeline` should squish redundant whitespace
+    run_ml_inference: boolean; -> Whether the `request_pipeline` should run the ML Inference pipeline
   }
   scheduling: {
-    enabled: boolean; -> Is sync schedule enabled?
+    enabled: boolean; -> Whether sync schedule is enabled
     interval: string; -> Quartz Cron syntax
   };
-  service_type: string; -> Optional, used for UI sugar
-  status: string;       -> Enum, see below
-  sync_now: boolean; -> Flag to signal user wants to initiate a sync
+  service_type: string; -> Service type of the connector
+  status: string;       -> Connector status Enum, see below
+  sync_now: boolean;    -> Flag to signal user wants to initiate a sync
 }
 ```
 **Possible values for 'status'**
-- `created` -> entry has been created but connector has not connected to elasticsearch (written by index creator)
-- `needs_configuration` -> connector has written its configuration to elasticsearch (written by connector)
-- `configured` -> connector has been fully configured (written by kibana on updating configuration, or directly by connector if no further configuration is necessary)
-- `connected` -> connector has successfully connected to the data source (written by connector on successfully connecting to data source )
-- `error` -> connector has encountered an error that needs recovery (written by connector on encountering any error)
+- `created` -> A document for a connector has been created in connector index but the connector has not connected to elasticsearch (written by index creator).
+- `needs_configuration` -> Configurable fields have been written into the connector, either by Kibana (for native connector) or connector (for custom connector.
+- `configured` -> A connector has been fully configured (written by Kibana on updating configuration, or directly by connector if no further configuration is necessary).
+- `connected` -> A connector has successfully connected to the data source (written by connector on successfully connecting to data source).
+- `error` -> A connector has encountered an error, either because the data source is not healthy or the last sync failed.
 
 **last_sync_status enum**
-- `null` -> no sync has ever started
-- `in_progress` -> sync successfully started
-- `completed` -> sync successfully completed
-- `error` -> sync error
+- `null` -> No sync job has ever started.
+- `in_progress` -> A sync job successfully started.
+- `completed` -> A sync job successfully completed.
+- `failed` -> A sync job failed.
 
-### Elasticsearch mappings for .elastic-connectors:
+#### Elasticsearch mappings for `.elastic-connectors`:
 ```
 "mappings" : {
   "properties" : {
-    "api_key_id" : { "type" : "keyword" },
```

Comment:
Hi @sphilipse , do you have any concern of removing this `api_key_id`? I see it's still added by Kibana.

Reply:
Yes, it's used by Kibana to keep track of the API key that has been issued and revoke it if the user issues a new one. Please keep it.